x-pack/test/cases_api_integration/security_and_spaces/tests/common/comments/find_comments.ts

```diff
@@ -81,33 +83,52 @@ export default ({ getService }: FtrProviderContext): void => {
       expect(caseComments.comments).to.eql(patchedCase.comments);
     });
 
-    it('should filter case comments', async () => {
+    it('should find only case comments of the correct type', async () => {
       const { body: postedCase } = await supertest
         .post(CASES_URL)
         .set('kbn-xsrf', 'true')
         .send(postCaseReq)
         .expect(200);
 
-      // post 2 comments
+      // post 5 comments of all possible types
       await supertest
         .post(`${CASES_URL}/${postedCase.id}/comments`)
         .set('kbn-xsrf', 'true')
         .send(postCommentUserReq)
         .expect(200);
 
-      const { body: patchedCase } = await supertest
+      await supertest
         .post(`${CASES_URL}/${postedCase.id}/comments`)
         .set('kbn-xsrf', 'true')
-        .send({ ...postCommentUserReq, comment: 'unique' })
+        .send(postCommentAlertReq)
+        .expect(200);
+
+      await supertest
+        .post(`${CASES_URL}/${postedCase.id}/comments`)
+        .set('kbn-xsrf', 'true')
+        .send(postCommentActionsReq)
+        .expect(200);
+
+      await supertest
+        .post(`${CASES_URL}/${postedCase.id}/comments`)
+        .set('kbn-xsrf', 'true')
+        .send(postExternalReferenceESReq)
+        .expect(200);
+
+      await supertest
+        .post(`${CASES_URL}/${postedCase.id}/comments`)
+        .set('kbn-xsrf', 'true')
+        .send(persistableStateAttachment)
```

Comment:
You can use the `bulkCreateAttachments` helper function to create the comments with one API call.

Reply:
If we use bulk create attachments we can't guarantee the order. So we'll need to be careful how we check for the type of the first created comment. One option would be to use the `arraysToEqual` helper and just ensure that all 5 attachments exist in the response.

Reply:
> If we use bulk create attachments we can't guarantee the order. 

Good point @jonathan-buttner but here there is no problem because we are only looking for 1 of these 5 attachments 😉